packages/clerk-js/src/ui/utils/useFormControl.ts

```diff
@@ -177,82 +168,28 @@ export const buildRequest = (fieldStates: Array<FormControlStateLike>): Record<s
-    errorText: string;
-    warningText: string;
-    successfulText: string;
-    isFocused: boolean;
-    informationText: string | LocalizationKey;
+    feedback: string;
+    feedbackType: FeedbackType;
   };
 };
 
 type DebouncingOption = {
-  hasLostFocus: boolean;
-  warningText: string | undefined;
-  errorText: string | undefined;
-  enableErrorAfterBlur: boolean | undefined;
-  successfulText: string | undefined;
-  isFocused: boolean;
-  informationText: LocalizationKey | string | undefined;
-  hasPassedComplexity: boolean;
-  skipBlur?: boolean;
+  feedback?: string;
+  feedbackType?: FeedbackType;
+  isFocused?: boolean;
   delayInMs?: number;
 };
-export const useFormControlFeedback = (opts: DebouncingOption): DebouncedFeedback => {
-  const {
-    hasLostFocus = false,
-    errorText = '',
-    warningText = '',
-    enableErrorAfterBlur = false,
-    successfulText = '',
-    isFocused = false,
-    informationText = '',
-    skipBlur = false,
-    delayInMs = 100,
-  } = opts;
-
-  const canDisplayFeedback = useMemo(() => {
-    if (enableErrorAfterBlur) {
-      if (skipBlur) {
-        return true;
-      }
-      return hasLostFocus;
-    }
-    return true;
-  }, [enableErrorAfterBlur, hasLostFocus, skipBlur]);
+export const useFormControlFeedback = (opts?: DebouncingOption): DebouncedFeedback => {
+  const { feedback = '', delayInMs = 100, feedbackType = 'info', isFocused = false } = opts || {};
 
-  const feedbackMemo = useMemo(() => {
-    const shouldDisplayErrorAsWarning = errorText && !skipBlur;
-    const _errorText = !shouldDisplayErrorAsWarning && canDisplayFeedback ? errorText : '';
-    const _warningText = shouldDisplayErrorAsWarning ? errorText : warningText;
-    const _successfulText = successfulText;
+  const debouncedFocused = useDebounce(isFocused, DEBOUNCE_MS);
 
-    /*
-     * On keyboard navigation avoid displaying the information text when an error is present.
-     * This is necessary in order to ensure that users will still be able to see the error message
-     *  even if they have pressed Enter (to submit form) and field still has focus.
-     */
-    const shouldShowInformationText = skipBlur
-      ? isFocused && !_successfulText && !_errorText
-      : isFocused && !_successfulText;
-    return {
-      errorText: _errorText,
-      successfulText: _successfulText,
-      warningText: _warningText,
-      isFocused,
-      informationText: shouldShowInformationText ? informationText : '',
-    };
-  }, [
-    informationText,
-    enableErrorAfterBlur,
-    isFocused,
-    successfulText,
-    hasLostFocus,
-    errorText,
-    canDisplayFeedback,
-    skipBlur,
-  ]);
+  const shouldHide = debouncedFocused ? false : ['info', 'warning'].includes(feedbackType);
 
-  const debouncedState = useSetTimeout(feedbackMemo, delayInMs);
+  const debouncedState = useDebounce(
+    { feedback: shouldHide ? '' : feedback, feedbackType: shouldHide ? 'info' : feedbackType },
+    delayInMs,
+  );
```

Comment:
Are there two layers of debouncing now with the addition of debouncing to the validate on change? Or are these two separate?

Reply:
There is debouncing on form feedback, which is not specific to the password field. That's set at 100 ms by default. We have added a debounce layer to the password validation for performance among other reasons. This means we have 2 layers when calling the `validate` function.